processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -378,21 +380,33 @@ public boolean isAssignableTo(Type other) {
                     modifiableGetters.put( Executables.getPropertyName( getter ), getter );
                 }
             }
-            getters = Collections.unmodifiableMap( modifiableGetters );
+
+            //TODO decide if we want to have a mechanism for turning on/off direct field mapping
```

Comment:
I think I'd give preference to calling getters/setters and fall back to field access if they aren't there and the field is accessible for the generated code (either public or protected and living in the same package).